docs/guides/3_validating_public_input.md

```diff
@@ -0,0 +1,201 @@
+# Validating public input
+
+In some applications, it is crucial to ensure that a third party has performed a computation correctly and to make use of the result of that computation. To achieve this, the third party must first interact with Aligned and obtain the `AlignedVerificationData`, a receipt indicating that the proof of the computation was verified correctly. The application should then receive both the `AlignedVerificationData` and the result of the computation. After confirming that the proof was verified by Aligned, it must check that the posted result matches the one committed in the `AlignedVerificationData`.
+
+This guide demonstrates how to validate a Risc0 proof using the Aligned SDK. The Risc0 program in this example is a Fibonacci sequence calculator. It generates a public input that corresponds to the last two Fibonacci numbers of the sequence, taken modulo 7919. Our goal is to validate, within a smart contract, that the public input commitments match these two numbers.
+
+In this case, the Fibonacci number to be calculated is **500** and the last two numbers of the sequence modulo 7919 are **1268** and **1926**.
+
+This guide assumes you are in the `examples/validating-public-input` directory.
+
+## Generate your ZK Proof
+
+To submit proofs to Aligned and get them verified, first you need to generate those proofs. Every proving system has its own way of generating proofs.
+
+You can find examples on how to generate proofs in the [generating proofs guide](4_generating_proofs.md).
+
+To generate the proof needed to try this example, run `make generate_risc_zero_fibonacci_proof`.
+
+## Write your smart contract
+
+To check if a proof was verified in Aligned, you need to make a call to the `AlignedServiceManager` contract inside your smart contract.
+
+Also, you will need a way to check that the proven program is the correct one.
+
+The Aligned CLI provides a way for you to get the verification key commitment without actually generating and submitting a proof.
```

Comment:
though technically correct, the term `verification key` here appears without much context and is confusing for a user without deep zk technical knowledge. Let's explain what the term refers to or just remove it

Reply:
Solved in [b6f3356](https://github.com/yetanotherco/aligned_layer/pull/648/commits/b6f335654a66bfc92eeba9b60f605509bad3ec0b)